provider/kubernetes/kubernetes_test.go

```diff
@@ -797,6 +798,166 @@ func TestIngressAnnotations(t *testing.T) {
 	assert.Equal(t, expected, actual)
 }
 
+func TestIngressClassAnnotation(t *testing.T) {
+	ingresses := []*v1beta1.Ingress{
+		buildIngress(
+			iNamespace("testing"),
+			iAnnotation(label.TraefikFrontendPassHostHeader, "true"),
```

Comment:
Do we need this for our particular test? How does it relate to validating the Ingress class logic?